sentry-android-core/src/main/java/io/sentry/android/core/SystemEventsBreadcrumbsIntegration.java

```diff
@@ -0,0 +1,197 @@
+    actions.add(ACTION_SHUTDOWN);
+    actions.add(ACTION_AIRPLANE_MODE_CHANGED);
+    actions.add(ACTION_BATTERY_LOW);
+    actions.add(ACTION_BATTERY_OKAY);
+    actions.add(ACTION_BOOT_COMPLETED);
+    actions.add(ACTION_CAMERA_BUTTON);
+    actions.add(ACTION_CONFIGURATION_CHANGED);
+    actions.add("android.intent.action.CONTENT_CHANGED");
+    actions.add(ACTION_DATE_CHANGED);
+    actions.add(ACTION_DEVICE_STORAGE_LOW);
+    actions.add(ACTION_DEVICE_STORAGE_OK);
+    actions.add(ACTION_DOCK_EVENT);
+    actions.add("android.intent.action.DREAMING_STARTED");
+    actions.add("android.intent.action.DREAMING_STOPPED");
+    actions.add(ACTION_INPUT_METHOD_CHANGED);
+    actions.add(ACTION_LOCALE_CHANGED);
+    actions.add(ACTION_REBOOT);
+    actions.add(ACTION_SCREEN_OFF);
+    actions.add(ACTION_SCREEN_ON);
+    actions.add(ACTION_TIMEZONE_CHANGED);
+    actions.add(ACTION_TIME_CHANGED);
+    actions.add("android.os.action.DEVICE_IDLE_MODE_CHANGED");
+    actions.add("android.os.action.POWER_SAVE_MODE_CHANGED");
+    // The user pressed the "Report" button in the crash/ANR dialog.
+    actions.add(ACTION_APP_ERROR);
+    // Show activity for reporting a bug.
+    actions.add(ACTION_BUG_REPORT);
+
+    // consider if somebody mounted or ejected a sdcard
+    actions.add(ACTION_MEDIA_BAD_REMOVAL);
+    actions.add(ACTION_MEDIA_MOUNTED);
+    actions.add(ACTION_MEDIA_UNMOUNTABLE);
+    actions.add(ACTION_MEDIA_UNMOUNTED);
+
+    return actions;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (receiver != null) {
+      context.unregisterReceiver(receiver);
+
+      if (options != null) {
+        options.getLogger().log(SentryLevel.DEBUG, "SystemEventsBreadcrumbsIntegration remove.");
+      }
+    }
+  }
+
+  private static final class SystemEventsBroadcastReceiver extends BroadcastReceiver {
+
+    private final @NotNull IHub hub;
+
+    SystemEventsBroadcastReceiver(final @NotNull IHub hub) {
+      this.hub = hub;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+      final Breadcrumb breadcrumb = new Breadcrumb();
+      breadcrumb.setType("info");
+      breadcrumb.setCategory("app.broadcast");
+      if (intent.getAction() != null) {
+        final int lastDotIndex = intent.getAction().lastIndexOf(".");
+        breadcrumb.setData("action", intent.getAction().substring(lastDotIndex + 1));
+      }
+
+      // TODO: should we log the params? sometimes its necessary eg
+      // ACTION_AIRPLANE_MODE_CHANGED
+      // its a single action with 2 states that differs on extras
+      final Bundle extras = intent.getExtras();
+      final Map<String, String> newExtras = new HashMap<>();
+      if (extras != null && !extras.isEmpty()) {
+        for (String item : extras.keySet()) {
+          try {
+            newExtras.put(item, extras.get(item).toString());
+          } catch (Exception ignored) {
+          }
+        }
+      }
+      breadcrumb.setData("extras", newExtras);
```

Comment:
this is trick, each broadcast has its own extras, we could send them all as I do now, but there are a lot of garbage too.
the other way would be to only send the useful extras for each broadcast.
the thing is, docs for extras are not great and it'd take some time to find the way keys.